### PR TITLE
feat(retention): change default policy, add API to apply retention

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java [hunk 1]
```diff
@@ -986,6 +988,27 @@ private UpdateAspectResult patchAspect(final Patch patch, final SystemMetadata s
...
+    if (start == null) {
+      start = 0;
+    }
+    args.start = start;
+    if (count == null) {
+      count = 100;
+    }
+    args.count = count;
+    if (attemptWithVersion == null) {
+      attemptWithVersion = 21;
```
Comment (pedro93) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:1003]: Where does this value come from? What is the reason for this default?

File: metadata-service/war/src/main/resources/boot/retention.yaml [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
-#  config:
-#    retention:
-#      version:
-#        maxVersions: 20
+- entity: "*"
+  aspect: "*"
+  config:
+    retention:
+      version:
+        maxVersions: 20
```
Comment (pedro93) [inline metadata-service/war/src/main/resources/boot/retention.yaml:12]: Can this be overridden by environment variables?

Comment (anshbansal) [reply metadata-service/war/src/main/resources/boot/retention.yaml:12]: No this is not supported currently.

File: metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java (hunk 1 above)
Comment (anshbansal) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java:1003]: From `maxVersions: 20` in `/boot/retention.yml` 20 max versions + 1 default to 0.

**Review by pedro93 — APPROVED** [review]

**Pull request merged**